src/test/java/teammates/test/cases/browsertests/AdminHomePageUiTest.java

```diff
@@ -86,12 +86,12 @@ private void testCreateInstructorAction() throws Exception {
         BackDoor.deleteCourse(demoCourseId);
         BackDoor.deleteInstructor(demoCourseId, instructor.email);
         homePage.createInstructorByInstructorDetailsSingleLineForm(instructorDetails);
```

Comment:
The calls to `RetryManager` can always be written in the test itself; just that I felt these few methods should be extracted into `BaseTestCaseWithBackDoorApiAccess` as they can be reused.

As for having to create multiple `*WithRetry` variants for methods, I feel that it's fine, as code reuse is already quite substantial. Have a look at this example:

```
InstructorAttributes instructor = RetryManager.runWithRetry(
new RetryableTaskWithResult<InstructorAttributes>("getInstructor") {
    @Override
    public boolean run() {
        setResult(getInstructor(courseId, instructorEmail));
        return getResult() != null;
    }
});
```

This anonymous class, in just a few lines, contains code specifying how to

1. Run the task
1. Check whether the task succeeded
1. Return the result

These 3 things are unique to the task being retried, and have to be specified for every single task. The success condition for `getInstructor` is that the return value is not null, whereas that for `getKeyForInstructor` is that the returned string starts with `[BACKDOOR_STATUS_FAILURE]` - one cannot write a general method that works for both `getInstructor` and `getKeyForInstructor` without specifying the success condition for each.

Having said that, I realise we can make retry declarations even cleaner and more readable by building in a library of common success conditions (not null, true, no assertion errors caught etc.), and by setting the result through the return value of the run code (instead of calling setResult). The same task above can possibly be written like this:

```
InstructorAttributes instructor = RetryManager.runUntilNotNull(
new RetryableTaskWithResult<InstructorAttributes>("getInstructor") {
    @Override
    public InstructorAttributes run() {
        return getInstructor(courseId, instructorEmail);
    }
});
```

More complex conditions can continue to be specified within the code, like this:

```
String key = RetryManager.runUntilSuccessful(
new RetryableTaskWithResult<String>("getKeyForInstructor") {
    @Override
    public String run() {
        return getKeyForInstructor(courseId, instructorEmail);
    }
    @Override
    public boolean isSuccessful(String result) {
        return !result.startsWith(Const.StatusCodes.BACKDOOR_STATUS_FAILURE);
    }
});
```

For tasks without a result, it could look like this:

```
RetryManager.runUntilNoException(
new RetryableTaskThrows<AssertionError>("HTML verification") {
    @Override
    public void run() throws AssertionError {
        verifyHtmlPart(MAIN_CONTENT, filePath);
    }
});
```

Let me know what you think of this idea! :)